server/server.go

```diff
@@ -810,7 +813,17 @@ func (c *client) run(ctx context.Context, run int, s *session, ke *messages.KE)
 
 	log.Printf("recv(%v) SR Run:%d DCMix:%x", c.raddr(), sr.Run, sr.DCMix)
 
+	var dcMixTotal int
+	for _, dcMix := range sr.DCMix {
+		dcMixTotal += len(dcMix)
```

Comment:
this check isn't sufficient. you must check that the outer slice has length c.pr.MessageCount, and each of the inner slices have length s.mtot.  It's a matrix with dimensions mcount * mtot.